server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeMeasureVariationsStep.java

```diff
@@ -126,35 +128,19 @@ private MeasuresWithVariationRepository computeMeasuresWithVariations(Component
       return measuresWithVariationRepository;
     }
 
-    private void processMeasuresWithVariation(Component component, MeasuresWithVariationRepository measuresWithVariationRepository) {
-      for (MeasureWithVariations measureWithVariations : measuresWithVariationRepository.measures()) {
-        Metric metric = measureWithVariations.getMetric();
-        Measure measure = Measure.updatedMeasureBuilder(measureWithVariations.getMeasure())
-          .setVariations(new MeasureVariations(
-            measureWithVariations.getVariation(1),
-            measureWithVariations.getVariation(2),
-            measureWithVariations.getVariation(3),
-            measureWithVariations.getVariation(4),
-            measureWithVariations.getVariation(5)))
-          .create();
-        measureRepository.update(component, metric, measure);
-      }
-    }
-
     private void setVariationMeasures(Component component, List<PastMeasureDto> pastMeasures, int period, MeasuresWithVariationRepository measuresWithVariationRepository) {
       Map<MeasureKey, PastMeasureDto> pastMeasuresByMeasureKey = from(pastMeasures).uniqueIndex(pastMeasureToMeasureKey);
-      for (Map.Entry<String, Measure> entry : from(measureRepository.getRawMeasures(component).entries()).filter(NotDeveloperMeasure.INSTANCE)) {
```

Comment:
No more indeed ! It was previously required when using measureRepository.getRawMeasures(component) because it's also returning dev measures, but measureRepository.getRawMeasure(component, metric) does not return dev measure, so I've been able to reduce the number of checks.
Moreover, using measureRepository.getRawMeasures(component) was returning far more measures that we want (it was returning dev measures and also measures on metric that we don't care, as we're only needing numeric metrics).